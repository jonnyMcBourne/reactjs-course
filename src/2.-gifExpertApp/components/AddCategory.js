import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({ setCategory }) => {

  const [inputState, setInputState] = useState("");
  
  const onChangeCategory = (e) => {
    console.log("E",e.target.value);
    const { target } = e;
    setInputState(target.value);
  };
  console.log('innnnnp',inputState);
  const addCategory = (e) => { 
    e.preventDefault();
    console.log("inputSTATE",inputState);
    if(inputState !== '' ){
      setCategory(inputState)
    }else{
      setInputState("");
    }
  };
  
return (
    <div>
      <p>{inputState}</p>
      <form onSubmit={addCategory}>

          <input
            type="text"
            id="category"
            name="category"
            placeholder="placeholder"
            onChange={onChangeCategory}
            value={inputState}
          />

        <input type="submit" name="addbutton" value="Add" />

      </form>
    </div>
  );
};

AddCategory.propTypes={
  setCategory: PropTypes.func.isRequired,
}