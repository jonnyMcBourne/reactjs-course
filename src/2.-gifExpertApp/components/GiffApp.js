import React from 'react'

export const GiffApp = () => {
    return (
        <div>
            hello
        </div>
    )
}
