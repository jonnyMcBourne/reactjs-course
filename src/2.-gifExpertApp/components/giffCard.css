.container {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 20%;
    height: 25vh;
    margin-right: 1.5%;
    margin-bottom: 1.5%;
    margin-top: 1.5%;
    margin-left: 1.5%;
    padding-top: 5%;
    border-radius: 20px;
    overflow: hidden;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}
.img{
    position: relative;
    bottom: 20%;
    width: 95%;
    height: 30vh;
    border-radius: 5px;
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
}
.title{
    height: 10% ;
    text-align: center;
    justify-content: center;
    align-items: center;
    position: relative;
    bottom: 10%;
}

  /* RESPONSIVE DESIGN*/
  
  /*cellphone Xl*/
  @media screen and (min-width: 260px) {
    .container {
        width: 80%;
        height: 25vh;
        margin-right: 1.5%;
        margin-bottom: 1.5%;
        margin-top: 1.5%;
        margin-left: 1.5%;
        padding-top: 5%;

    }

}

/*cellphone Xl*/
@media screen and (min-width: 480px) {

}

/*Tablet*/
@media screen and (min-width: 767px) {
    .container {
        width: 40%;
        height: 25vh;
        margin-right: 1.5%;
        margin-bottom: 1.5%;
        margin-top: 1.5%;
        margin-left: 1.5%;
        padding-top: 5%;
    }

}

@media screen and (min-width: 920px) {
    .container {
        width: 30%;
        height: 25vh;
        margin-right: 1.5%;
        margin-bottom: 1.5%;
        margin-top: 1.5%;
        margin-left: 1.5%;
        padding-top: 5%;
    }

}

/*Web*/
@media screen and (min-width: 1020px) {
    .container {
        width: 20%;
        height: 25vh;
        margin-right: 1.5%;
        margin-bottom: 1.5%;
        margin-top: 1.5%;
        margin-left: 1.5%;
        padding-top: 5%;
    }

}