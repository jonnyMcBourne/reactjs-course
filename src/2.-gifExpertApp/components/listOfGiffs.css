.listContainer{
    display: flex;
    width: 100%;
    flex-wrap: wrap;
    justify-content: center;
}


  
  /* RESPONSIVE DESIGN*/
  
  /*cellphone Xl*/
  @media screen and (min-width: 260px) {

  }
  
  /*cellphone Xl*/
  @media screen and (min-width: 480px) {

  }
  
  /*Tablet*/
  @media screen and (min-width: 767px) {

  }
  
  /*Web*/
  @media screen and (min-width: 950px) {

  }