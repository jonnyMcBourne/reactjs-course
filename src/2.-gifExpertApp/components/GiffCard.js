import React from 'react'
import './giffCard.css'
export const GiffCard = ({giff}) => {
    const {title,img}= giff
    return (
      <span className="container">
        <div style={{backgroundImage:`url(${img})`}} className="img" >
        </div>
        <h4 className="title" >{title}</h4>
      </span>
    );
}
