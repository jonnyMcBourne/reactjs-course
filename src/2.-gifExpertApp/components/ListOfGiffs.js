import React from 'react'
import { Fragment } from 'react/cjs/react.production.min'
import {GiffCard} from './GiffCard'
import './listOfGiffs.css'

export const ListOfGiffs = ({categories=[]}) => {
    return (
      <div className="listContainer">
        {categories.map((giff) => (
            <Fragment key={giff.id}>
            <GiffCard giff={giff} />
            </Fragment>
        ))}
      </div>
    );
}
