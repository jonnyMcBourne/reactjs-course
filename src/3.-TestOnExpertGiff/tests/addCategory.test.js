import React from 'react';
import { shallow } from 'enzyme';
import {AddCategory} from '../../2.-gifExpertApp/components/AddCategory';
import {GiffApp} from '../../2.-gifExpertApp/components/GiffApp';


describe('TEST ON COMPONENT ADDCATEGORY',()=>{
    //Global Variables
        const wrapperglobal = shallow(<GiffApp/>);
        
        const setCategory = jest.fn()
        let wrapperCategory = shallow(<AddCategory setCategory={setCategory}/>)
        
        const changeInput=()=>{
            //arrange
            const input=wrapperCategory.find('input').at(0)
            const value ='hello world';
                //act
            input.simulate('change',{target:{value:value}});
        }

        beforeEach(()=>{
            //jest.clearAllMocks();
            wrapperCategory = shallow(<AddCategory setCategory={setCategory}/>)
        })

    test('test on main component renders properly',()=>{
        expect(wrapperglobal).toMatchSnapshot();
    });
    
    test('text input shoould change',()=>{
            //arrange
        const input=wrapperCategory.find('input').at(0);
        const value ='hello world';
            //act
        input.simulate('change',{target:{value:value}});
            //assert
        expect(wrapperCategory.find('p').text().trim() ).toBe(value)
    });


    test('shouldnt submit',()=>{
        //arrange
        const form=wrapperCategory.find('form');
        //act
        form.simulate('submit', { preventDefault(){}  });
        //assert
        expect(setCategory).not.toHaveBeenCalled();
    });

    test('should submit',()=>{
        //arrange
        const input=wrapperCategory.find('input').at(0);
        const value ='hello world';
        const form=wrapperCategory.find('form');
        //act
        input.simulate('change',{target:{value:value}});
        form.simulate('submit', { preventDefault(){}  });
        //assert
        expect(setCategory).toHaveBeenCalled();
    });



    
    
});