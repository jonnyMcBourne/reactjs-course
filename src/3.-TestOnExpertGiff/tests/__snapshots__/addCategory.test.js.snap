// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST ON COMPONENT ADDCATEGORY test on main component renders properly 1`] = `
<div>
  <h1>
    GiffExpert
  </h1>
  <hr />
  <AddCategory
    setCategory={[Function]}
  />
  <ListOfGiffs
    categories={Array []}
  />
</div>
`;
